protocols/ping/CHANGELOG.md

```diff
@@ -1,3 +1,13 @@
+# Unreleased
+
+- Don't close connection if ping protocol is unsupported by remote.
+  Previously, a failed protocol negotation for ping caused a force close of the connection.
+  As a result, all nodes in a network had to support ping.
+  To allow networks where some nodes don't support ping, we now emit
+  `PingFailure::Unsupported` once for every connection on which ping is not supported.
```

Comment:
```suggestion
  `PingFailure::Unsupported` once for every connection on which ping is not supported.
  
  In case you want to stick with the old behavior, you need to close the connection
  manually on `PingFailure::Unsupported`.
```